server/src/services/metadata.service.ts

```diff
@@ -487,6 +496,124 @@ export class MetadataService implements OnEvents {
     }
   }
 
+  async searchPerson(ownerId: AssetEntity['ownerId'], dto: SearchPeopleDto): Promise<PersonResponseDto[]> {
+    return this.personRepository.getByName(ownerId, dto.name, { withHidden: dto.withHidden });
+  }
+
+  private async applyTaggedFaces(asset: AssetEntity, tags: ImmichTags) {
+    this.logger.debug(`Starting faces extraction from tags (${asset.id})`);
+
+    const faceSet = new Set();
+
+    if (!tags.RegionInfo) {
+      return true;
+    }
+
+    interface Region {
+      imageWidth: number;
+      imageHeight: number;
+      boundingBoxX1: number;
+      boundingBoxX2: number;
+      boundingBoxY1: number;
+      boundingBoxY2: number;
+    }
+
+    interface MetadataFaceResult {
+      Name?: string;
+      Type?: string;
+      Region: Region;
+    }
+
+    function toPixels(region: Region, unit: string): Region {
+      const pixelRegion: Region = region;
+      if (unit.toLowerCase() == 'normalized') {
+        pixelRegion.boundingBoxX1 = Math.floor(region.boundingBoxX1 * region.imageWidth);
+        pixelRegion.boundingBoxX2 = Math.floor(region.boundingBoxX2 * region.imageWidth);
+        pixelRegion.boundingBoxY1 = Math.floor(region.boundingBoxY1 * region.imageHeight);
+        pixelRegion.boundingBoxY2 = Math.floor(region.boundingBoxY2 * region.imageHeight);
+      }
+      return pixelRegion;
+    }
+
+    for (const region of tags.RegionInfo?.RegionList) {
+      const faceResult: MetadataFaceResult = {
+        Name: region.Name,
+        Type: region.Type,
+        Region: toPixels(
+          {
+            imageWidth: tags.RegionInfo?.AppliedToDimensions?.W,
+            imageHeight: tags.RegionInfo?.AppliedToDimensions?.H,
+            boundingBoxX1: region.Area.X - region.Area.W / 2,
+            boundingBoxY1: region.Area.Y - region.Area.H / 2,
+            boundingBoxX2: region.Area.X + region.Area.W / 2,
+            boundingBoxY2: region.Area.Y + region.Area.H / 2,
+          },
+          region.Area.Unit,
+        ),
+      };
+
+      faceSet.add(faceResult);
```

Comment:
Adding different objects to a set will always add a new entry to the set. That being said, does it really happen that the same face can be duplicated in this metadata?

Reply:
AFAIK metadata's regions are a list, it is responsibility of the tagging application avoid adding duplicates. I used a set here to be cautious about that possibility.

Reply:
You can use the bounding box dimensions as the key to the set to make this work.

Reply:
Hope it's ok leaving it this way, if not please let me know